### PR TITLE
Code rev

### PR DESCRIPTION
**Pull request by osama-maree**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,7 +1,33 @@
...
+import { RouterProvider, createBrowserRouter } from "react-router-dom";
+import PageNotFound from "./pages/PageNotFound.jsx";
+import Home from "./pages/Home.jsx";
+import { Box } from "@mui/material";
+import Layout from "./component/Layout.jsx";
+import Order from "./pages/Order.jsx";
 
+const router = createBrowserRouter([
+  {
+    path: "",
```
Comment (m5atib) [inline src/App.js]:
```suggestion
    path: "/",
```

File: src/component/Layout.jsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import NavBar from "./NavBar.jsx";
+import { Outlet } from "react-router-dom";
+
+const Layout = () => {
+  return (
+    <>
+      <NavBar />
+      <Outlet></Outlet>
```
Comment (m5atib) [inline src/component/Layout.jsx]:
```suggestion
      <Outlet />
```

File: src/component/ModalForm.jsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  width: 400,
+  bgcolor: "background.paper",
+  border: "2px solid #00acc1",
+  boxShadow: 24,
+  p: 4,
+};
+
+export default function TransitionsModal({ handleClose, open }) {
+  const handleSubmit = (values, { resetForm }) => {
+    console.log("Submitted values:", values);
```
Comment (m5atib) [inline src/component/ModalForm.jsx]: Please remove any `console.log`

File: src/pages/Order.jsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+const Order = () => {
+  const [open, setOpen] = React.useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+  const { order, setOrder } = useShopping();
+
+  const handleRemove = (id) => {
+    setOrder(order.filter((ord) => ord.id !== id));
+  };
+  let TotalAmmount = order.reduce((accumulator, product) => {
```
Comment (m5atib) [inline src/pages/Order.jsx]:
```suggestion
  const totalAmount = order.reduce((accumulator, product) => {
```

File: src/pages/Order.jsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        </Typography>
+        <Button
+          variant="outlined"
+          sx={{ color: "#00acc1", border: "solid 1px #00acc1" }}
+        >
+          Total Price: {TotalAmmount}
+        </Button>
+      </Box>
+      <Divider />
+      {order.length > 0 ? (
```
Comment (m5atib) [inline src/pages/Order.jsx:48]:
```suggestion
      {!!order.length ? (
```

File: src/component/ProductListItem.jsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+          <IconButton edge="end" aria-label="delete" onClick={()=>handleRemove(id)}>
+            <DeleteIcon color="error" />
+          </IconButton>
+        </ListItemSecondaryAction>
+      </ListItem>
+    </List>
+  );
+};
+
+export default ProductListItem;
```
Comment (m5atib) [inline src/component/ProductListItem.jsx]: we can optimize the code by using the correct `React.memo` condition (if you render this when its quantity changed)

File: src/component/ProductListItem.jsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import {
+  ListItem,
+  ListItemText,
+  IconButton,
+  List,
+  ListItemSecondaryAction,
+  ImageListItem,
+  Typography,
+} from "@mui/material";
```
Comment (m5atib) [inline src/component/ProductListItem.jsx:10]: this is correct by the bottom is the optimized version of importing dependencies
```suggestion
import  ListItem from "@mui/material/ListItem "
import  ListItemText from "@mui/material/ListItemText "
import  IconButton from "@mui/material/IconButton "
import  List from "@mui/material/List "
import  ListItemSecondaryAction from "@mui/material/ListItemSecondaryAction "
import  ImageListItem from "@mui/material/ImageListItem "
import Typography from "@mui/material/Typography";
```

File: src/App.js [hunk 8]
```diff
@@ -1,7 +1,33 @@
-import "./App.css";
+import { RouterProvider, createBrowserRouter } from "react-router-dom";
+import PageNotFound from "./pages/PageNotFound.jsx";
+import Home from "./pages/Home.jsx";
```
Comment (m5atib) [inline src/App.js]: can you please use the ` React.lazy ` loading

File: src/context/ShoppingContext.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+        setOrder,
+      }}
+    >
+      {children}
+    </ShoppingContext.Provider>
+  );
+};
+export const useShopping = () => {
+  return useContext(ShoppingContext);
+};
```
Comment (m5atib) [inline src/context/ShoppingContext.js]: please separate this hook in another file and lets add more validation on the code like this :

```suggestion
export default function useShopping() {
  const context = useContext(ShoppingContext);

  if (!context) {
    throw new Error(
      "useShopping must be used within a ShoppingContextProvider"
    );
  }

  return context;
}
```

File: src/component/Pagination.jsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import { Pagination, Stack } from "@mui/material";
+
+const itemsPerPage = 12;
+
+const PaginationComponent = ({ totalItems, currentPage, onPageChange }) => {
```
Comment (m5atib) [inline src/component/Pagination.jsx]: you should never called the component SomethingComponent
you should give it a name with linking it with Component
so this must be 

```suggestion
const Pagination = ({ totalItems, currentPage, onPageChange }) => {
```

File: src/component/ProductListItem.jsx [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+            src={`/assets/${image}`}
+            alt={name}
+            sx={{ borderRadius: "100px !important" }}
+            loading="lazy"
+          />
+        </ImageListItem>
+        <ListItemText
+          primary={name}
+          secondary={
+            <React.Fragment>
```
Comment (m5atib) [inline src/component/ProductListItem.jsx:48]: you can use also `<>`  its the same with `<React.Fragment>`

`<> ... </>`

File: src/component/ShoppingCart.jsx [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+import { useShopping } from "../context/ShoppingContext.js";
+
+const ShoppingCart = ({ products }) => {
+  const { setOrder, order } = useShopping();
+  const [currentPage, setCurrentPage] = useState(1);
+  const [product, setProduct] = useState(products);
+  const itemsPerPage = 12;
+  const startIndex = (currentPage - 1) * itemsPerPage;
+  const endIndex = startIndex + itemsPerPage;
+  let currentData = product.slice(startIndex, endIndex);
```
Comment (m5atib) [inline src/component/ShoppingCart.jsx]: please try as possible to avoid `let ` and keep using `const `

File: src/component/ShoppingCart.jsx [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+  const handleQuantityDec = (index) => {
+    setProduct(
+      product.map((prod) =>
+        prod.id === index
+          ? { ...prod, quantity: prod.quantity == 0 ? 0 : prod.quantity - 1 }
+          : { ...prod }
+      )
+    );
+  };
+  const handleAddToCart = (index) => {
```
Comment (m5atib) [inline src/component/ShoppingCart.jsx:37]: can you find a more optimized data structure to store the products in a cart?

File: src/component/ModalForm.jsx [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+  email: Yup.string()
+    .email("Invalid email address")
+    .required("Email is required"),
+});
+
+const initialValues = {
+  name: "",
+  email: "",
+};
+const style = {
```
Comment (HamzehSaleh) [inline src/component/ModalForm.jsx:25]: As you are using MUI, why you are not using `makeStyle` & `createStyle`

**Review by m5atib — CHANGES_REQUESTED** [review]
Please deploy your app and make the needed changes

File: src/component/ModalForm.jsx [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+  left: "50%",
+  transform: "translate(-50%, -50%)",
+  width: 400,
+  bgcolor: "background.paper",
+  border: "2px solid #00acc1",
+  boxShadow: 24,
+  p: 4,
+};
+
+export default function TransitionsModal({ handleClose, open }) {
```
Comment (HamzehSaleh) [inline src/component/ModalForm.jsx]: `open ` is a boolean so we should add a prefix to it like ( is, has, can, ... etc)
`handleClose `is called a callback function, so when receiving a function let's put "on" with the event name.
```suggestion
export default function TransitionsModal({ onClose, isOpen }) {
```

File: src/component/ModalForm.jsx [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+
+                <div>
+                  <Button
+                    type="submit"
+                    variant="contained"
+                    sx={{
+                      backgroundColor: "#00acc1",
+                      "&:hover": { backgroundColor: "#00acc1" },
+                      marginTop: "1rem",
+                      width: "100%",
```
Comment (HamzehSaleh) [inline src/component/ModalForm.jsx:101]: we prefer using classes with more than 3 properties

File: src/pages/PageNotFound.jsx [hunk 17]
```diff
@@ -0,0 +1,129 @@
+import React from "react";
+import "../style/pagenotfound.css";
+const PageNotFound = () => {
```
Comment (HamzehSaleh) [inline src/pages/PageNotFound.jsx:3]: Read about[ Error Boundaries](https://legacy.reactjs.org/docs/error-boundaries.html)  to make a generic solution

File: src/component/product.js [hunk 18]
```diff
@@ -0,0 +1,205 @@
+export const products = [
```
Comment (HamzehSaleh) [inline src/component/product.js]: As this is not a component, I prefer moving it outside this folder

File: src/component/ShoppingCart.jsx (hunk 13 above)
Comment (osama-maree) [reply src/component/ShoppingCart.jsx:37]: Of course, in this case we are dealing with a product that is impossible to dauplicat. It is possible to use the map to increase performance, key maybe Uniqe as Name or ID from production

File: src/component/ModalForm.jsx (hunk 14 above)
Comment (osama-maree) [reply src/component/ModalForm.jsx:25]: well, I will use it in the next task

**Pull request merged**